com.unity.mobile.notifications/Runtime/Android/Plugins/com/unity/androidnotifications/UnityNotificationUtilities.java

```diff
@@ -72,43 +72,44 @@ protected static int findResourceIdInContextByName(Context context, String name)
        Unfortunately, while Notification itself is Parcelable and can be marshalled to bytes,
        it's contents are not guaranteed to be (Binder objects).
        Hence what we try to do here is:
-       - serialize as is
-       - fallback 1: serialize our known properties + serialize extras as is
-       - fallback 2: serialize our known stuff
-       When notification is serialized as-is, it may contain references to resources and in case
-       of app update may fail to deserialize due to resources now missing, hence always save fallback version.
+       - serialize as is if notification is possibly customized by user
+       - otherwise serialize our stuff, since there is nothing more
     */
-    protected static void serializeNotification(SharedPreferences prefs, Notification notification) {
+    protected static void serializeNotification(SharedPreferences prefs, Notification notification, boolean serializeParcel) {
         try {
-            String serialized = null, fallback = null;
+            String serialized;
             ByteArrayOutputStream data = new ByteArrayOutputStream();
             DataOutputStream out = new DataOutputStream(data);
-            if (serializeNotificationCustom(notification, out)) {
-                out.flush();
-                byte[] bytes = data.toByteArray();
-                fallback = Base64.encodeToString(bytes, 0, bytes.length, 0);
+            if (serializeParcel) {
+                Intent intent = new Intent();
+                intent.putExtra(KEY_NOTIFICATION, notification);
+                if (serializeNotificationParcel(intent, out)) {
+                    out.close();
+                    byte[] bytes = data.toByteArray();
+                    serialized = Base64.encodeToString(bytes, 0, bytes.length, 0);
+                } else {
+                    return; // failed
```

Comment:
shouldn't we log it, like you do below Log.e(TAG_UNITY, "Failed to serialize notification", e); ?

Reply:
serializeNotificationParcel already logs it, so this would be a second logging for the same error.